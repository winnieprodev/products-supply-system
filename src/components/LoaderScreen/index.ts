export { default as LoaderScreen } from "./LoaderScreen";
