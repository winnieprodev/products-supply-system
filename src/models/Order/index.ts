export * from "./enums";
export * from "./Order.d";
