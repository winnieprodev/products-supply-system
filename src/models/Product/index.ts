export * from "./Product.d";
export * from "./enums";
