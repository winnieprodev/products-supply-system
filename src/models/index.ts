export * from "./Cart";
export * from "./Order";
export * from "./Outlet";
export * from "./Product";
export * from "./Search";
export * from "./Staff";
export * from "./Supplier";
export * from "./User";
export * from "./utils";
