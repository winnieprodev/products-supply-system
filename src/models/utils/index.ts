export * from "./async.d";
export * from "./enums";
