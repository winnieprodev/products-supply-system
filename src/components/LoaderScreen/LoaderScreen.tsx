import { WrapperMode } from "models";
import React from "react";

import { Loader } from "../Loader";
import * as Styled from "./loaderScreen.styled";

interface OwnProps {
  mode?: WrapperMode;
}
export const LoaderScreen: React.FC<React.PropsWithChildren<OwnProps>> = ({
  children,
  mode = WrapperMode.full,
}) => (
  <Styled.Wrapper $mode={mode}>
    <Loader />
    <Styled.Message>{children}</Styled.Message>
  </Styled.Wrapper>
);

export default React.memo(LoaderScreen);
