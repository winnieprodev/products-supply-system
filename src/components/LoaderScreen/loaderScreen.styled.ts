import { WrapperMode } from "models";
import styled, { css, keyframes } from "styled-components";

interface WrapperProps {
  $mode: WrapperMode;
}

const popTinyWrapper = keyframes`
  from{ 
    transform: scale(0.2);
    opacity: 0;
  }
  to {
    opacity: 1;
    transform: scale(0.5);
  }
`;

export const Wrapper = styled.div<WrapperProps>`
  ${({ $mode }) =>
    $mode === WrapperMode.full
      ? css`
          margin: 20vh auto;
        `
      : css`
          margin: 8px auto;
        `}

  width: 60%;
  display: flex;
  flex-direction: column;
  align-items: center;

  ${({ $mode }) =>
    $mode === WrapperMode.tiny &&
    css`
      width: 48px;
      position: fixed;
      right: 30px;
      top: -4px;
      transform: scale(0.5);
      animation: ${popTinyWrapper} ease 0.3s;
    `};
`;

export const Message = styled.div`
  font-weight: bold;
  margin-top: 16px;
`;
