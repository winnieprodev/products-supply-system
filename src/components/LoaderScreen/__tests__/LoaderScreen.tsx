import { render } from "@testing-library/react";
import { WrapperMode } from "models";

import LoaderScreen from "../LoaderScreen";

describe("LoaderScreen", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(<LoaderScreen>test</LoaderScreen>);
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for mode=singleLine", () => {
      const { container } = render(
        <LoaderScreen mode={WrapperMode.singleLine}>test</LoaderScreen>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for mode=tiny", () => {
      const { container } = render(
        <LoaderScreen mode={WrapperMode.tiny}>test</LoaderScreen>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
