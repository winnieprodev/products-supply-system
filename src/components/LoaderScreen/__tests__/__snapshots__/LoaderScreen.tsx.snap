// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderScreen Snapshot tests it matches the snapshot 1`] = `
.c1 {
  width: 48px;
  height: 48px;
  padding: 8px;
  position: relative;
}

.c1:before {
  position: absolute;
  right: 0;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: -1;
  background: #F3F3F3;
  display: block;
  content: "";
  border-radius: 50%;
  -webkit-animation: hAYQBa 5s infinite;
  animation: hAYQBa 5s infinite;
}

.c2 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c3 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.4s;
  animation-delay: 0.4s;
  opacity: 0.3;
}

.c4 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.7s;
  animation-delay: 0.7s;
  opacity: 0.3;
}

.c0 {
  margin: 20vh auto;
  width: 60%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-weight: bold;
  margin-top: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      class="c2"
      src="logo.svg"
    />
    <img
      class="c3"
      src="logo.svg"
    />
    <img
      class="c4"
      src="logo.svg"
    />
  </div>
  <div
    class="c5"
  >
    test
  </div>
</div>
`;

exports[`LoaderScreen Snapshot tests it matches the snapshot for mode=singleLine 1`] = `
.c1 {
  width: 48px;
  height: 48px;
  padding: 8px;
  position: relative;
}

.c1:before {
  position: absolute;
  right: 0;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: -1;
  background: #F3F3F3;
  display: block;
  content: "";
  border-radius: 50%;
  -webkit-animation: hAYQBa 5s infinite;
  animation: hAYQBa 5s infinite;
}

.c2 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c3 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.4s;
  animation-delay: 0.4s;
  opacity: 0.3;
}

.c4 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.7s;
  animation-delay: 0.7s;
  opacity: 0.3;
}

.c0 {
  margin: 8px auto;
  width: 60%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-weight: bold;
  margin-top: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      class="c2"
      src="logo.svg"
    />
    <img
      class="c3"
      src="logo.svg"
    />
    <img
      class="c4"
      src="logo.svg"
    />
  </div>
  <div
    class="c5"
  >
    test
  </div>
</div>
`;

exports[`LoaderScreen Snapshot tests it matches the snapshot for mode=tiny 1`] = `
.c1 {
  width: 48px;
  height: 48px;
  padding: 8px;
  position: relative;
}

.c1:before {
  position: absolute;
  right: 0;
  top: 0;
  left: 0;
  bottom: 0;
  z-index: -1;
  background: #F3F3F3;
  display: block;
  content: "";
  border-radius: 50%;
  -webkit-animation: hAYQBa 5s infinite;
  animation: hAYQBa 5s infinite;
}

.c2 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c3 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.4s;
  animation-delay: 0.4s;
  opacity: 0.3;
}

.c4 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: 48px;
  height: 48px;
  -webkit-animation: iizYDp 10s infinite;
  animation: iizYDp 10s infinite;
  -webkit-animation-delay: 0.7s;
  animation-delay: 0.7s;
  opacity: 0.3;
}

.c0 {
  margin: 8px auto;
  width: 60%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 48px;
  position: fixed;
  right: 30px;
  top: -4px;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
  -webkit-animation: fewUyK ease 0.3s;
  animation: fewUyK ease 0.3s;
}

.c5 {
  font-weight: bold;
  margin-top: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      class="c2"
      src="logo.svg"
    />
    <img
      class="c3"
      src="logo.svg"
    />
    <img
      class="c4"
      src="logo.svg"
    />
  </div>
  <div
    class="c5"
  >
    test
  </div>
</div>
`;
